kratos/includes/mesh.h

```diff
@@ -60,143 +53,116 @@ namespace Kratos
 ///@name Kratos Classes
 ///@{
 
-/// Mesh is the second level of abstraction in the data structure which hold Nodes, Elements and Conditions and their Properties.
 /**
- * Mesh is the second level of abstraction in the data structure which hold Nodes, Elements and Conditions and their Properties.
+ * @class Mesh
+ * @ingroup KratosCore
+ * @brief Mesh is the second level of abstraction in the data structure which hold Nodes, Elements and Conditions and their Properties.
+ * @details Mesh is the second level of abstraction in the data structure which hold Nodes, Elements and Conditions and their Properties.
  * In other words, Mesh is a complete pack of all type of entities without any additional data associated with them.
  * So a set of Elements and Conditions with their Nodes and Properties can be grouped together as a Mesh and send to
  * procedures like mesh refinement, material optimization, mesh movement or any other procedure which works on entities
  * without needing additional data for their processes.
+ * @author Pooyan Dadvand
 */
 template<class TNodeType, class TPropertiesType, class TElementType, class TConditionType>
 class Mesh : public DataValueContainer, public Flags
 {
 public:
-
-
     ///@name Type Definitions
     ///@{
 
     /// Pointer definition of Mesh
     KRATOS_CLASS_POINTER_DEFINITION(Mesh);
 
-    typedef std::size_t IndexType;
-
-    typedef std::size_t SizeType;
-
-    typedef TNodeType NodeType;
-
-    typedef TPropertiesType PropertiesType;
-
-    typedef Geometry<NodeType> GeometryType;
-
-    typedef TElementType ElementType;
-
-    typedef TConditionType ConditionType;
-
-    typedef MasterSlaveConstraint MasterSlaveConstraintType;
-
-    typedef Mesh<TNodeType, TPropertiesType, TElementType, TConditionType> MeshType;
+    // Alias for representing indices, typically used for indexing arrays or collections.
+    using IndexType = std::size_t;
+
+    // Alias for representing sizes or counts, commonly used to indicate the size of data structures.
+    using SizeType = std::size_t;
+
+    // Alias for representing node types in a mesh data structure.
+    using NodeType = TNodeType;
```

Comment:
We should delete this one an use Node directly.

Reply:
It is a template argument, IMHO we could remove all those templates or directly inegrate the mesh into the model part

Reply:
Agree